### PR TITLE
[BugFix] Fix show create catalog that does not exist

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -2491,15 +2491,20 @@ private List<List<String>> doPredicate(ShowStmt showStmt,
         return returnRows;
     }
 
-    private void handleShowCreateExternalCatalog() {
+    private void handleShowCreateExternalCatalog() throws AnalysisException {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:2494]: Better to add a sql-test case to cover this case.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:2494]: +1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|3|66.67%|[2505]|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:2494]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|3|66.67%|[2505]|

File: fe/fe-core/src/test/java/com/starrocks/qe/ShowExecutorTest.java [hunk 2]
```diff
@@ -1157,6 +1157,20 @@ public Catalog getCatalogByName(String name) {
...
+        new MockUp<CatalogMgr>() {
+            @Mock
+            public Catalog getCatalogByName(String name) {
+                return null;
+            }
+        };
+
+        ShowCreateExternalCatalogStmt stmt = new ShowCreateExternalCatalogStmt("catalog_not_exist");
+        ShowExecutor executor = new ShowExecutor(ctx, stmt);
+        Assert.assertThrows(AnalysisException.class, executor::execute);
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/qe/ShowExecutorTest.java:1171]: nit: use `ExceptionChecker.expectThrowsWithMsg`

Comment (letian-jiang) [reply fe/fe-core/src/test/java/com/starrocks/qe/ShowExecutorTest.java]: updated

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|3|66.67%|[2505]|

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**